src/runtime_src/core/edge/user/graph_object.cpp

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+#include "graph_object.h"
+#include "core/edge/user/aie/graph.h"
+#include "core/common/system.h"
+#include <memory>
+
+namespace zynqaie {
+  graph_object::graph_object(ZYNQ::shim* shim, const xrt::uuid& uuid , const char* name,
+                  xrt::graph::access_mode am, const zynqaie::hwctx_object* hwctx)
+    : m_shim{shim}
+  {
+    auto device{xrt_core::get_userpf_device(m_shim)};
+    std::string gname{name};
+    m_graphInstance = std::make_unique<zynqaie::graph_instance>(device, gname, am, hwctx,uuid);
+  }
+
+  void
+  graph_object::reset_graph()
+  {
+    try
+    {
+      m_graphInstance->reset();
+    }
+    catch (const std::exception& e)
+    {
+      xrt_core::send_exception_message(std::string("fail to reset graph: ") + e.what());
+    }
+    catch (...)
+    {
+      xrt_core::send_exception_message(std::string("fail to reset graph"));
+    }
```

Comment:
As for formatting.  Not sure about what convention is used throughout this file, but I prefer the braces on same line as condition or try/catch.
```suggestion
    try {
      m_graphInstance->reset();
    }
    catch (const std::exception& e)  {
      xrt_core::send_exception_message(std::string("fail to reset graph: ") + e.what());
    }
    catch (...) {
      xrt_core::send_exception_message(std::string("fail to reset graph"));
    }
```